subprojects/gradle/gradle-logger/src/main/kotlin/com/avito/logger/LoggerRegistry.kt

```diff
@@ -0,0 +1,103 @@
+package com.avito.logger
+
+import com.avito.android.elastic.ElasticConfig
+import com.avito.android.sentry.SentryConfig
+import com.avito.logger.destination.ElasticDestination
+import com.avito.logger.destination.SentryDestination
+import com.avito.logger.destination.Slf4jDestination
+import com.avito.logger.formatter.AppendMetadataFormatter
+import com.avito.logger.handler.CombinedHandler
+import com.avito.logger.handler.DefaultLoggingHandler
+import com.avito.logger.handler.LoggingHandler
+
+object LoggerRegistry {
+
+    private val loggersCache = mutableMapOf<String, Logger>()
+
+    fun create(
+        isCiRun: Boolean,
+        className: String,
+        projectPath: String,
+        sentryConfig: SentryConfig,
+        elasticConfig: ElasticConfig,
+        pluginName: String? = null,
+        taskName: String? = null
+    ): Logger = provideLogger(
+        isCiRun = isCiRun,
+        sentryConfig = sentryConfig,
+        elasticConfig = elasticConfig,
+        metadata = LoggerMetadata(
+            className = className,
+            pluginName = pluginName,
+            projectPath = projectPath,
+            taskName = taskName
+        )
+    )
+
+    private fun provideLogger(
+        isCiRun: Boolean,
+        sentryConfig: SentryConfig,
+        elasticConfig: ElasticConfig,
+        metadata: LoggerMetadata
+    ): Logger = if (isCiRun) {
+        loggersCache.getOrPut(metadata.className) {
```

Comment:
I expect that metadata will be a key for that cache

Reply:
i seriously don't know why we cache here; 
removed it for now, because fear of weird bugs

Reply:
As for me cache needed only for Okhttp reusing under the ElasticLogger